src/SFA.DAS.EmployerFinance.MessageHandlers/CommandHandlers/ProcessPeriodEndPaymentsCommandHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using NServiceBus;
+using SFA.DAS.EmployerFinance.Messages.Commands;
+using SFA.DAS.EmployerFinance.Queries.GetAllEmployerAccounts;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EmployerFinance.MessageHandlers.CommandHandlers
+{
+    public class ProcessPeriodPaymentsEndCommandHandler : IHandleMessages<ProcessPeriodEndPaymentsCommand>
+    {
+        private readonly IMediator _mediator;
+        private readonly ILog _logger;
+
+        public ProcessPeriodPaymentsEndCommandHandler(IMediator mediator, ILog logger)
+        {
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        public async Task Handle(ProcessPeriodEndPaymentsCommand message, IMessageHandlerContext context)
+        {
+            var response = await _mediator.SendAsync(new GetAllEmployerAccountsRequest());
+
+            var tasks = new List<Task>();
+
+            foreach (var account in response.Accounts)
+            {
+                _logger.Info($"Creating payment queue message for account ID: '{account.Id}' period end ref: '{message.PeriodEndRef}'");
+
+                tasks.Add(context.SendLocal<ImportAccountPaymentsCommand>(c =>
```

Comment:
There'll potentially be tens of thousands of messages (one for each account) so the outbox might be a bottleneck here so it might be worth doing something along the lines of:

```c#
var sendOptions = new SendOptions();

sendOptions.RouteToThisEndpoint();
sendOptions.RequireImmediateDispatch(); // Circumvent sender outbox
sendOptions.SetMessageId($"{message.PeriodEndRef}-{account.Id}"); // Allow receiver outbox to de-dupe

context.Send(new ImportAccountPaymentsCommand { PeriodEndRef = message.PeriodEndRef, AccountId = account.Id }, sendOptions);
```

Reply:
@cofaulco Updated my comment, probably worth including the name of the message in the ID to ensure it's unique.